play-json/shared/src/main/scala/Reads.scala

```diff
@@ -398,8 +398,57 @@ trait DefaultReads extends LowPriorityDefaultReads {
   /**
    * Deserializer for Array[T] types.
    */
-  implicit def ArrayReads[T: Reads: ClassTag]: Reads[Array[T]] = new Reads[Array[T]] {
-    def reads(json: JsValue) = json.validate[List[T]].map(_.toArray)
+  implicit def ArrayReads[T: Reads: ClassTag]: Reads[Array[T]] =
+    Reads[Array[T]] { _.validate[List[T]].map(_.toArray) }
+
+  // If the number is decimal, considering it's /1000 of millis,
+  // otherwise considering it's directly in millis
+  private def numberDuration(n: BigDecimal): FiniteDuration = {
+    val millis = if (!n.ulp.isWhole) {
+      (n.toDouble * 1000).toLong
+    } else n.toLong
+
+    FiniteDuration(millis, "ms")
+  }
+
+  /**
+   * Deserializer for Scala FiniteDuration
+   */
+  implicit val finiteDurationWrites: Reads[FiniteDuration] =
+    Reads[FiniteDuration] {
+      case JsNumber(n) => JsSuccess(numberDuration(n))
+      case JsString("0") => JsSuccess(Duration.Zero)
```

Comment:
This doesn't need to be a special case. Let's just write with the units "0 days".